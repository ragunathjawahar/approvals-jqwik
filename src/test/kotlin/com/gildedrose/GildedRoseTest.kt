package com.gildedrose

import net.jqwik.api.Arbitraries
import net.jqwik.api.Arbitrary
import net.jqwik.api.ForAll
import net.jqwik.api.From
import net.jqwik.api.Property
import net.jqwik.api.Provide
import org.approvaltests.Approvals

class GildedRoseTest {
  companion object {
    const val DATA_SET_SIZE = 50
  }

  @Property(
    tries = 1,
    seed = "-8556729318537930054"
  )
  fun `Aged Brie`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Aged Brie"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "1002985723690070170"
  )
  fun `Backstage passes to a TAFKAL80ETC concert`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Backstage passes to a TAFKAL80ETC concert"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "3398327011427988732"
  )
  fun `Sulfuras, Hand of Ragnaros`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Sulfuras, Hand of Ragnaros"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "6350692656906137395"
  )
  fun `all other items`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "all other items"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Provide
  fun sellIns(): Arbitrary<List<Int>> =
    Arbitraries
      .integers()
      .between(0, 50)
      .list()
      .uniqueElements()
      .ofSize(DATA_SET_SIZE)

  @Provide
  fun qualities(): Arbitrary<List<Int>> =
    Arbitraries
      .integers()
      .between(-10, 60)
      .list()
      .uniqueElements()
      .ofSize(DATA_SET_SIZE)
}
