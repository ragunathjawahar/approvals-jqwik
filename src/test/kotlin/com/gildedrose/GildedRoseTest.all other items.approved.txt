all other items, 31, 11
all other items, 0, -7
all other items, 20, -5
all other items, 11, -2
all other items, -1, 49
all other items, 4, 21
all other items, 33, 59
all other items, 12, 43
all other items, 10, -3
all other items, 6, 2
all other items, 15, 24
all other items, 36, 37
all other items, 29, 1
all other items, 40, 13
all other items, 48, -10
all other items, 21, 6
all other items, 26, -8
all other items, 2, 49
all other items, 35, -9
all other items, 49, 55
all other items, 1, 7
all other items, 42, 0
all other items, 28, 25
all other items, 3, 58
all other items, 14, 16
all other items, 13, 52
all other items, 8, 3
all other items, 19, 15
all other items, 34, 23
all other items, 32, 5
all other items, 5, 35
all other items, 22, 54
all other items, 23, 0
all other items, 7, 12
all other items, 37, 17
all other items, 9, 32
all other items, 46, 10
all other items, 43, -4
all other items, 30, 9
all other items, 24, -6
all other items, 45, 28
all other items, 41, 53
all other items, 17, 19
all other items, 47, 48
all other items, 39, -1
all other items, 25, 26
all other items, 38, 41
all other items, 18, 51
all other items, 44, 4
all other items, 27, 44
